### PR TITLE
#116 auditing for tables

### PR DESCRIPTION
**Pull request by tliszyns**

### CODE REVIEW

**Comment by tliszyns** [comment]
Why travis build failed? First I've got error `'dependencies.dependency.version' for io.oasp.java.modules:devon4j-jpa-envers:jar is missing`. After I add version in pom I've got:
`Could not resolve dependencies for project com.devonfw.java.mtsj:mtsj-core:jar:1.12.3-SNAPSHOT: Could not find artifact io.oasp.java.modules:devon4j-jpa-envers:jar:3.0.2 in central (https://repo.maven.apache.org/maven2)`

File: java/mtsj/core/src/main/resources/db/specific/h2/V0001__R001_Create_schema.sql [hunk 1]
```diff
@@ -199,4 +201,64 @@ CREATE TABLE RevInfo(
   id BIGINT NOT NULL GENERATED BY DEFAULT AS IDENTITY (START WITH 1),
   "timestamp" BIGINT NOT NULL,
   userLogin VARCHAR(255)
+);
+
+-- *** Category_AUD ***
+CREATE CACHED TABLE PUBLIC.Category_AUD(
```
Comment (hohwille) [inline java/mtsj/core/src/main/resources/db/specific/h2/V0001__R001_Create_schema.sql]: For homogenity we should either qualify all tables with schema or none. Same applies for `CACHED` keyword that is omitted in all existing create table statements. We should do that consistently within the DDL.

**Review by hohwille — CHANGES_REQUESTED** [review]
@tliszyns thanks for this PR. This looks quite good 👍 
I just added a few comments for improvement.
Some more questions:
* Dont we need the *_AUD tables for all supported DB types? Wouldn't otherwise MTS break for all other DBs?
* Did we pick the right entities/tables for auditing? IMHO all looks perfect. The only one I could question is the blob table. The orders/bookings are create "immutable" so it is perfectly correct that they are not audited.

**Comment by tliszyns** [comment]
> * Dont we need the *_AUD tables for all supported DB types? Wouldn't otherwise MTS break for all other DBs?
There are AUD tables for other DB types.

**Pull request closed without merging**